benchopt/utils/parquet.py

```diff
@@ -0,0 +1,62 @@
+
+import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+JSON_KEY = b"_meta_json"
+
+
+def to_parquet(df, path, metadata=None):
+    """Write a pandas.DataFrame in a parquet file, with optional metadata.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        DataFrame to write in the parquet file.
+    path: str | Path
+        Path to write the parquet file.
+    metadata: dict or None
+        Metadata to store in the parquet file. This metadata should be
+        serializable with json.
+    """
+    table = pa.Table.from_pandas(df)
+    new_metadata = {
+        JSON_KEY: json.dumps(metadata).encode("utf-8"),
+        **table.schema.metadata
+    }
+    table = table.replace_schema_metadata(new_metadata)
+    pq.write_table(table, path)
+
+
+def update_metadata(path, metadata):
+    """Update metadata in an existing parquet file.
+
+    Parameters
+    ----------
+    path: str | Path
+        Path of the parquet file to update.
+    metadata: dict or None
+        Metadata to store in the parquet file. This metadata should be
+        serializable with json.
```

Comment:
```suggestion
        serializable with ``json``.
```

Reply:
This is the protocol name Json, not the module so I would not put quotes here.